src/function-builder.ts

```diff
@@ -120,6 +121,61 @@ function assertRuntimeOptionsValid(runtimeOptions: RuntimeOptions): boolean {
       `serviceAccount must be set to 'default', a service account email, or '{serviceAccountName}@'`
     );
   }
+
```

Comment:
Should we explicitly disallow implicitly reserved labels? e.g. I think we reserve the use of `deployment-tools` (as in, we are going to overwrite whatever you put as `deployment-tools` with whatever the CLI wants it to be.

Reply:
I like this. I'd also possibly disable any label that starts with `firebase-`